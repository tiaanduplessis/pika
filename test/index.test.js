import pika from '../src'

test('should export function', () => {
  expect(pika).toBeDefined()
  expect(typeof pika).toBe('function')
})

test('should select DOM elements', () => {
  document.body.innerHTML = `<div class="foo" >
      <span id="baz" />
      <button id="bar" />
    </div>`

  expect(pika('body')).toMatchSnapshot()
  expect(pika('head')).toMatchSnapshot()
  expect(pika('div')).toMatchSnapshot()
  expect(pika('.foo')).toMatchSnapshot()
  expect(pika('#baz')).toMatchSnapshot()
  expect(pika('span')).toMatchSnapshot()
})
