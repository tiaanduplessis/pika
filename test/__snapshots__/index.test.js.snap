// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should select DOM elements 1`] = `
<body>
  <div
    class="foo"
  >
    
      
    <span
      id="baz"
    >
      
      
      <button
        id="bar"
      >
        
    
      </button>
    </span>
  </div>
</body>
`;

exports[`should select DOM elements 2`] = `<head />`;

exports[`should select DOM elements 3`] = `
Array [
  <div
    class="foo"
  >
    
      
    <span
      id="baz"
    >
      
      
      <button
        id="bar"
      >
        
    
      </button>
    </span>
  </div>,
]
`;

exports[`should select DOM elements 4`] = `
Array [
  <div
    class="foo"
  >
    
      
    <span
      id="baz"
    >
      
      
      <button
        id="bar"
      >
        
    
      </button>
    </span>
  </div>,
]
`;

exports[`should select DOM elements 5`] = `
Array [
  <span
    id="baz"
  >
    
      
    <button
      id="bar"
    >
      
    
    </button>
  </span>,
]
`;

exports[`should select DOM elements 6`] = `
Array [
  <span
    id="baz"
  >
    
      
    <button
      id="bar"
    >
      
    
    </button>
  </span>,
]
`;
